Realm/RLMObjectStore.mm

```diff
@@ -242,7 +242,9 @@ static bool RLMRealmCreateTables(RLMRealm *realm, RLMSchema *targetSchema, bool
 
         // validate versions
         if (oldVersion > newVersion && oldVersion != RLMNotVersioned) {
-            @throw RLMException(@"Version of Realm file on disk is higher than current schema version", @{@"path" : realm.path});
+            NSString *reason = [NSString stringWithFormat:@"Version (%lu) of Realm file on disk (%@) is higher than current schema version (%lu)",
```

Comment:
Maybe rephrase to:
`Realm version number %lu is greater than the current schema version %lu for Realm at path '%@'
Seems weird to have the path between the two numbers in the message.